lib/modules/manager/gradle/extract/consistent-versions-plugin.ts

```diff
@@ -0,0 +1,225 @@
+        currentVersion: versionAndPosition.version,
+        lockedVersion: lockFileMap.get(propDep)?.version,
+        depType: lockFileMap.get(propDep)?.depType,
+      } as PackageDependency<GradleManagerData>;
+      extractedDeps.push(newDep);
+      // Remove from the lockfile map so the same exact lib will not be included in globbing
+      lockFileMap.delete(propDep);
+    }
+  }
+
+  // For each regular expression dep in props file (starting with the longest glob string)...
+  for (const [propDepGlob, propVerAndPos] of propsFileRegexMap) {
+    const globRegex = globToRegex(propDepGlob);
+    for (const [exactDep, lockVersionAndDepType] of lockFileMap) {
+      if (globRegex.test(exactDep)) {
+        const newDep: Record<string, any> = {
+          managerData: {
+            packageFile: VERSIONS_PROPS,
+            fileReplacePosition: propVerAndPos.filePos,
+          },
+          depName: exactDep,
+          currentValue: propVerAndPos.version,
+          currentVersion: propVerAndPos.version,
+          lockedVersion: lockVersionAndDepType.version,
+          depType: lockVersionAndDepType.depType,
+          groupName: propDepGlob,
+        } as PackageDependency<GradleManagerData>;
+        extractedDeps.push(newDep);
+        // Remove from the lockfile map so the same exact lib will not be included in globbing
+        lockFileMap.delete(exactDep);
+      }
+    }
+  }
+  return extractedDeps;
+}
+
+// Translate glob syntax to a regex that does the same
+// Loosely borrowed mapping to regex from https://github.com/palantir/gradle-consistent-versions/blob/develop/src/main/java/com/palantir/gradle/versions/FuzzyPatternResolver.java
+function globToRegex(depName: string): RegExp {
+  return regEx(
+    depName
+      .replaceAll('*', '_WC_CHAR_')
+      .replace(/[/\-\\^$*+?.()|[\]{}]/g, '\\$&')
+      .replaceAll('_WC_CHAR_', '.*?')
+  );
+}
+
+interface VersionWithPosition {
+  version: string;
+  filePos: number;
+}
+
+interface VersionWithDepType {
+  version: string;
+  depType: string;
+}
+
+/**
+ * Parses `versions.lock`
+ */
+export function parseLockFile(input: string): Map<string, VersionWithDepType> {
+  const lockLineRegex = regEx(
+    `^(?<depName>[^:]+:[^:]+):(?<lockVersion>[^ ]+) \\(\\d+ constraints: [0-9a-f]+\\)$`
+  );
+
+  const depVerMap = new Map<string, VersionWithDepType>();
+  let isTestDepType = false;
+  for (const line of input.split(newlineRegex)) {
+    const lineMatch = lockLineRegex.exec(line);
+    if (lineMatch?.groups) {
+      const { depName, lockVersion } = lineMatch.groups;
+      if (isDependencyString(`${depName}:${lockVersion}`)) {
+        depVerMap.set(depName, {
+          version: lockVersion,
+          depType: isTestDepType ? 'test' : 'dependencies',
+        } as VersionWithDepType);
+      }
+    } else if (line === '[Test dependencies]') {
+      isTestDepType = true; // We know that all lines below this header are test dependencies
```

Comment:
Added support for depType: 'test' by parsing the special header in lockfile - all deps below are test deps

Reply:
Don't know, perhaps `testDependencies` is more precise? If I understood correctly, you also need to add whatever string we end up with somewhere else in the project?

Reply:
I'd suggest to go with `test`, since this is already with `sbt` manager as well. You can set any value there, in theory, and it should be the only place where it needs to be specified. Users could then define a package rule specifying `matchDepType` and customize behavior.